### PR TITLE
Configurable tracing policy in DeliveryOptions

### LINKED ISSUE #3737: Tracing and span are not propagated and recorded when EventBusProxies or EventBus is used to communicate between multiple microservices
### Version

4.0.0

### Context

Testing zipkin tracing repository, I discovered that the traceId and spanId generated when an HTTP request is received on the server are not propagated and stored when communication between microservices are done through EventBusProxies/EventBus. 

### Steps to reproduce

1. create vertx project, enable tracing and import io.vertx:vertx-zipkin:4.0.0
2. send a request to vertx http server
3. forward request to microservices through EventBusProxies or Eventbus
4. monitor recorded traces, request sent through Eventbus will not be recorded

### PR DESCRIPTION
**Pull request by tsegismont**
Closes #3737

### CODE REVIEW

File: src/main/java/io/vertx/core/eventbus/impl/OutboundDeliveryContext.java [hunk 1]
```diff
@@ -118,7 +118,9 @@ public void next() {
         if (message.trace == null) {
           src = true;
           BiConsumer<String, String> biConsumer = (String key, String val) -> message.headers().set(key, val);
-          message.trace = tracer.sendRequest(ctx, SpanKind.RPC, TracingPolicy.PROPAGATE, message, message.send ? "send" : "publish", biConsumer, MessageTagExtractor.INSTANCE);
+          TracingPolicy tracingPolicy = options.getTracingPolicy();
+          if (tracingPolicy == null) tracingPolicy = TracingPolicy.PROPAGATE;
```
Comment (vietj) [inline src/main/java/io/vertx/core/eventbus/impl/OutboundDeliveryContext.java:122]: missing braces

**Review by vietj — COMMENTED** [review]
Minor commnet about missing curly braces.

We will need also docs changes in Zipkin and Opentracing, e.g https://vertx.io/docs/vertx-zipkin/java/#_tracing_policy

LGTM

**Comment by tsegismont** [comment]
Yes, a pull request for `vertx-tracing` parent is coming (docs+tests).

I got some doubts about the test in this PR, I may push another commit later. In this case, I'll ping you. Otherwise, I'll go ahead and merge.

File: src/main/java/io/vertx/core/eventbus/impl/OutboundDeliveryContext.java (hunk 1 above)
Comment (tsegismont) [reply src/main/java/io/vertx/core/eventbus/impl/OutboundDeliveryContext.java]: Done

**Comment by vietj** [comment]
it looks good for merging though

**Pull request merged**